File: app/controllers/katello/activation_keys_controller.rb
#
# Copyright 2013 Red Hat, Inc.
#
# This software is licensed to you under the GNU General Public
# License as published by the Free Software Foundation; either version
# 2 of the License (GPLv2) or (at your option) any later version.
# There is NO WARRANTY for this software, express or implied,
# including the implied warranties of MERCHANTABILITY,
# NON-INFRINGEMENT, or FITNESS FOR A PARTICULAR PURPOSE. You should
# have received a copy of GPLv2 along with this software; if not, see
# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.

module Katello
  class ActivationKeysController < Katello::ApplicationController
    respond_to :html, :js

    before_filter :authorize

    def rules
      read_test = lambda {ActivationKey.readable?(current_organization)}
      {
        :index => read_test,
        :all => read_test
      }
    end

    def title
      _('Activation Keys')
    end

    def index
      render 'bastion/layouts/application', :layout => false
    end

    def all
      redirect_to action: 'index', :anchor => '/activation-keys'
    end
  end
end

File: engines/bastion/app/assets/javascripts/bastion/activation-keys/activation-keys.controller.js
/**
 * Copyright 2013-2014 Red Hat, Inc.
 *
 * This software is licensed to you under the GNU General Public
 * License as published by the Free Software Foundation; either version
 * 2 of the License (GPLv2) or (at your option) any later version.
 * There is NO WARRANTY for this software, express or implied,
 * including the implied warranties of MERCHANTABILITY,
 * NON-INFRINGEMENT, or FITNESS FOR A PARTICULAR PURPOSE. You should
 * have received a copy of GPLv2 along with this software; if not, see
 * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
*/

/**
 * @ngdoc object
 * @name  Bastion.activation-keys.controller:ActivationKeysController
 *
 * @requires $scope
 * @requires $location
 * @requires translate
 * @requires Nutupane
 * @requires ActivationKey
 * @requires CurrentOrganization
 *
 * @description
 *   Provides the functionality specific to activation keys for use with the Nutupane UI pattern.
 *   Defines the columns to display and the transform function for how to generate each row
 *   within the table.
 */
angular.module('Bastion.activation-keys').controller('ActivationKeysController',
    ['$scope', '$location', 'translate', 'Nutupane', 'ActivationKey', 'CurrentOrganization',
    function ($scope, $location, translate, Nutupane, ActivationKey, CurrentOrganization) {

        var params = {
            'organization_id':  CurrentOrganization,
            'search':           $location.search().search || "",
            'order':            'name ASC',
            'paged':            true
        };

        var nutupane = new Nutupane(ActivationKey, params);
        $scope.table = nutupane.table;
        $scope.removeRow = nutupane.removeRow;

        $scope.table.closeItem = function () {
            $scope.transitionTo('activation-keys.index');
        };

    }]
);
